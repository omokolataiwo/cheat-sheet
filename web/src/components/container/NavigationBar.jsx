import React from 'react';
import { Link } from 'react-router-dom';
import GitLogo from '../../asset/git-logo.png';

const renderAdminNavBar = () => (
  <nav>
    <div className="nav-wrapper">
      <Link to="/" className="brand-logo">
        <img src={GitLogo} width="50" height="50" alt="Logo" />
      </Link>

      <ul id="nav-mobile" className="right hide-on-med-and-down">
        <li>
          <Link to="/" className="nav-link m-2 menu-item">
            Home
          </Link>
        </li>
        <li>
          <Link to="/user" className="nav-link m-2 menu-item">
            Dash Board
          </Link>
        </li>
        <li>
          <Link to="/logout" className="nav-link m-2 menu-item">
            Logout
          </Link>
        </li>
      </ul>
    </div>
  </nav>
);

const renderNavBar = () => (
  <nav>
    <div className="nav-wrapper">
      <Link to="/" className="brand-logo">
        <img src={GitLogo} width="50" height="50" alt="Logo" />
      </Link>
      <ul id="nav-mobile" className="right hide-on-med-and-down">
        <li>
          <Link to="/" className="nav-link m-2 menu-item">
            Home
          </Link>
        </li>
      </ul>
    </div>
  </nav>
);

export default ({ adminBar }) => (adminBar ? renderAdminNavBar() : renderNavBar());
