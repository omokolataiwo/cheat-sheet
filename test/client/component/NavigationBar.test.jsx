import React from 'react';
import { shallow } from 'enzyme';
import NavigationBar from '../../../web/src/components/container/NavigationBar';

let props = {
  cheat: { description: 'init repository', line: 'git init' },
  onAddToFavorite: jest.fn(() => {})
};
const wrapper = shallow(<NavigationBar {...props} />);

describe('Cheat Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render admin nav bar', () => {
    props = { ...props, adminBar: true };
    const wrapper = shallow(<NavigationBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
