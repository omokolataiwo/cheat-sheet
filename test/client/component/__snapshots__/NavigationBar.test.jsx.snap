// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheat Component should render admin nav bar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    adminBar={true}
    cheat={
      Object {
        "description": "init repository",
        "line": "git init",
      }
    }
    onAddToFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nav-wrapper"
      >
        <Link
          className="brand-logo"
          replace={false}
          to="/"
        >
          <img
            alt="Logo"
            height="50"
            src={Object {}}
            width="50"
          />
        </Link>
        <ul
          className="right hide-on-med-and-down"
          id="nav-mobile"
        >
          <li>
            <Link
              className="nav-link m-2 menu-item"
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>
          <li>
            <Link
              className="nav-link m-2 menu-item"
              replace={false}
              to="/user"
            >
              Dash Board
            </Link>
          </li>
          <li>
            <Link
              className="nav-link m-2 menu-item"
              replace={false}
              to="/logout"
            >
              Logout
            </Link>
          </li>
        </ul>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="brand-logo"
            replace={false}
            to="/"
          >
            <img
              alt="Logo"
              height="50"
              src={Object {}}
              width="50"
            />
          </Link>,
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
          >
            <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/user"
              >
                Dash Board
              </Link>
            </li>
            <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/logout"
              >
                Logout
              </Link>
            </li>
          </ul>,
        ],
        "className": "nav-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="Logo"
              height="50"
              src={Object {}}
              width="50"
            />,
            "className": "brand-logo",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Logo",
              "height": "50",
              "src": Object {},
              "width": "50",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>,
              <li>
                <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/user"
                >
                  Dash Board
                </Link>
              </li>,
              <li>
                <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/logout"
                >
                  Logout
                </Link>
              </li>,
            ],
            "className": "right hide-on-med-and-down",
            "id": "nav-mobile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "className": "nav-link m-2 menu-item",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/user"
                >
                  Dash Board
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Dash Board",
                  "className": "nav-link m-2 menu-item",
                  "replace": false,
                  "to": "/user",
                },
                "ref": null,
                "rendered": "Dash Board",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/logout"
                >
                  Logout
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Logout",
                  "className": "nav-link m-2 menu-item",
                  "replace": false,
                  "to": "/logout",
                },
                "ref": null,
                "rendered": "Logout",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper"
        >
          <Link
            className="brand-logo"
            replace={false}
            to="/"
          >
            <img
              alt="Logo"
              height="50"
              src={Object {}}
              width="50"
            />
          </Link>
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
          >
            <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/user"
              >
                Dash Board
              </Link>
            </li>
            <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/logout"
              >
                Logout
              </Link>
            </li>
          </ul>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="brand-logo"
              replace={false}
              to="/"
            >
              <img
                alt="Logo"
                height="50"
                src={Object {}}
                width="50"
              />
            </Link>,
            <ul
              className="right hide-on-med-and-down"
              id="nav-mobile"
            >
              <li>
                <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/user"
                >
                  Dash Board
                </Link>
              </li>
              <li>
                <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/logout"
                >
                  Logout
                </Link>
              </li>
            </ul>,
          ],
          "className": "nav-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Logo"
                height="50"
                src={Object {}}
                width="50"
              />,
              "className": "brand-logo",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Logo",
                "height": "50",
                "src": Object {},
                "width": "50",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                    className="nav-link m-2 menu-item"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>,
                <li>
                  <Link
                    className="nav-link m-2 menu-item"
                    replace={false}
                    to="/user"
                  >
                    Dash Board
                  </Link>
                </li>,
                <li>
                  <Link
                    className="nav-link m-2 menu-item"
                    replace={false}
                    to="/logout"
                  >
                    Logout
                  </Link>
                </li>,
              ],
              "className": "right hide-on-med-and-down",
              "id": "nav-mobile",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link m-2 menu-item"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "className": "nav-link m-2 menu-item",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link m-2 menu-item"
                    replace={false}
                    to="/user"
                  >
                    Dash Board
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Dash Board",
                    "className": "nav-link m-2 menu-item",
                    "replace": false,
                    "to": "/user",
                  },
                  "ref": null,
                  "rendered": "Dash Board",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link m-2 menu-item"
                    replace={false}
                    to="/logout"
                  >
                    Logout
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Logout",
                    "className": "nav-link m-2 menu-item",
                    "replace": false,
                    "to": "/logout",
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Cheat Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    cheat={
      Object {
        "description": "init repository",
        "line": "git init",
      }
    }
    onAddToFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nav-wrapper"
      >
        <Link
          className="brand-logo"
          replace={false}
          to="/"
        >
          <img
            alt="Logo"
            height="50"
            src={Object {}}
            width="50"
          />
        </Link>
        <ul
          className="right hide-on-med-and-down"
          id="nav-mobile"
        >
          <li>
            <Link
              className="nav-link m-2 menu-item"
              replace={false}
              to="/"
            >
              Home
            </Link>
          </li>
        </ul>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="brand-logo"
            replace={false}
            to="/"
          >
            <img
              alt="Logo"
              height="50"
              src={Object {}}
              width="50"
            />
          </Link>,
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
          >
            <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
          </ul>,
        ],
        "className": "nav-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="Logo"
              height="50"
              src={Object {}}
              width="50"
            />,
            "className": "brand-logo",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Logo",
              "height": "50",
              "src": Object {},
              "width": "50",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>,
            "className": "right hide-on-med-and-down",
            "id": "nav-mobile",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/"
              >
                Home
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "className": "nav-link m-2 menu-item",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper"
        >
          <Link
            className="brand-logo"
            replace={false}
            to="/"
          >
            <img
              alt="Logo"
              height="50"
              src={Object {}}
              width="50"
            />
          </Link>
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
          >
            <li>
              <Link
                className="nav-link m-2 menu-item"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
          </ul>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="brand-logo"
              replace={false}
              to="/"
            >
              <img
                alt="Logo"
                height="50"
                src={Object {}}
                width="50"
              />
            </Link>,
            <ul
              className="right hide-on-med-and-down"
              id="nav-mobile"
            >
              <li>
                <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
            </ul>,
          ],
          "className": "nav-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Logo"
                height="50"
                src={Object {}}
                width="50"
              />,
              "className": "brand-logo",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Logo",
                "height": "50",
                "src": Object {},
                "width": "50",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li>
                <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>,
              "className": "right hide-on-med-and-down",
              "id": "nav-mobile",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="nav-link m-2 menu-item"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "className": "nav-link m-2 menu-item",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
